UPGRADE.md

```diff
@@ -6,6 +6,16 @@ awareness about deprecated code.
 - Use of our low-overhead runtime deprecation API, details:
   https://github.com/doctrine/deprecations/
 
+# Upgrade to 3.2
+
+The usage of Doctrine Cache is deprecated in favor of PSR-6. The following
+methods related to Doctrine Cache have been replaced with PSR-6 counterparts:
+
+- `Configuration::setResultCacheImpl()` is deprecated, use `setResultCache()` instead.
+- `Configuration::getResultCacheImpl()` is deprecated, use `getResultCache()` instead.
+- `QueryCacheProfile::setResultCacheDriver()` is deprecated, use `setResultCache()` instead.
+- `QueryCacheProfile::getResultCacheDriver()` is deprecated, use `getResultCache()` instead.
```

Comment:
If none of these are in a hot path, it's ok to add the deprecations now in a single PR. If not, maybe let's postpone the deprecations to a subsequent minor release.

Reply:
Those deprecations are all connected. `Connection::executeCacheQuery()` needs a result cache and pulls it either from the passed `QueryCacheProfile` or (if the profile does not contain a cache implementation) from its own `Configuration` instance. It does not really make sense to switch only one of the classes to PSR-6.

Reply:
I don't understand your comment. What I was saying is we don't want to get too many unfixable deprecations. So if some or all of these can happen a lot during a single HTTP request or process, then we should probably add the new methods in a new minor, and then the deprecations in a subsequent minor. I'm discussing the same issue for `doctrine/persistence` with @alcaeus , but FYI we had that very issue during the namespace move we did in that library, and IIRC we vowed to always proceed in 2 phases as described above.

Reply:
Okay, so I would basically remove all deprecations and just add a second way of configuring a cache? Yes, I could do that.

Reply:
One question through: We could already soft-deprecate those methods by adding the `@deprecated` annotations, right? I would only remove the triggers, correct?

Reply:
Done!